### DISCUSSION TITLE
Treat warnings as errors

### DISCUSSION THREAD

**Discussion by fondesa** [androiddevelopersitalia/aditogether#19] [General]
Both Kotlin and Android provide flags to treat all the warnings as errors (e.g. Android lint warnings).
This means that, a warning which is not fixed or suppressed explicitly, results in a compilation failure.
I've used this successfully in both small and very large projects without any issues, improving the code quality overall since users have to take an explicit action over a warning.

What do you think @4face-studi0 @mcatta ?

**Comment by fardavide** [comment]
100% for it!

**Reply by mcatta** [reply]
Sounds good, if this improves the qualify go for it!

**Comment by fardavide** [comment]
Given we applied this to the project, I think we can lock the discussion
